test/unit/light/unit_MpegTs.cpp

```
@@ -0,0 +1,309 @@
+        if (pidOf(pkt) == mm::ts::kPidPmt && !pmt) pmt = section(pkt);
+    }
+    REQUIRE(pat != nullptr);
+    REQUIRE(pmt != nullptr);
+
+    // The PAT's first program entry names the PID its map lives on (13 bits; top 3 reserved).
+    const uint16_t mapPid = static_cast<uint16_t>(((pat[10] & 0x1F) << 8) | pat[11]);
+    CHECK(mapPid == mm::ts::kPidPmt);
+
+    // The PMT names the elementary stream: H.264 (type 0x1B) on the video PID.
+    const uint16_t progInfoLen = static_cast<uint16_t>(((pmt[10] & 0x0F) << 8) | pmt[11]);
+    const uint8_t* es = pmt + 12 + progInfoLen;
```

Cppcheck (2.21.0)

[error] Pointer addition with NULL pointer.

(nullPointerArithmetic)